### PR TITLE
Bottom Sheet example with new API

### PR DESCRIPTION
**Pull request by enterteg**
Adds bottom sheet example with new API. Need to be tested on Android.

### CODE REVIEW

File: examples/Example/src/showcase/bottomSheet/index.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ export class BottomSheet extends Component<
       { useNativeDriver: USE_NATIVE_DRIVER }
     );
     this.lastScrollY.addListener(({ value }) => {
+      console.log(value);
```
Comment (jakub-gonet) [inline examples/Example/src/showcase/bottomSheet/index.tsx:64]: Should we have `console.log` here?

File: examples/Example/src/new_api/chat_heads/index.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+const HEADER_HEIGHT = 44; // let's assume it's always the same for ease of understanding ;)
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/chat_heads/index.tsx:12]: Can't we get it from [navigation](https://reactnavigation.org/docs/elements/#useheaderheight)?

**Review by jakub-gonet — APPROVED** [review]
`examples/Example/src/new_api/drag_n_drop/index.tsx` should be removed.

File: examples/Example/src/new_api/chat_heads/index.tsx (hunk 2 above)
Comment (enterteg) [reply examples/Example/src/new_api/chat_heads/index.tsx]: It wasn't not part of this PR. Removed this file.

File: examples/Example/src/new_api/bottom_sheet/index.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    translationY.value = e.translationY;
+  };
+
+  const onHeaderHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
+
+  const onPanHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
```
Comment (jakub-gonet) [inline examples/Example/src/new_api/bottom_sheet/index.tsx:56]: Can't we inline those?

Comment (enterteg) [reply examples/Example/src/new_api/bottom_sheet/index.tsx:56]: Yeah previously logic was different, will update.

File: examples/Example/src/new_api/bottom_sheet/index.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        <LoremIpsum words={200} />
+        <Animated.View style={[styles.bottomSheet, bottomSheetAnimatedStyle]}>
+          <GestureDetector gesture={headerGesture}>
+            <View style={styles.header} />
+          </GestureDetector>
+          <GestureDetector
+            gesture={Gesture.Simultaneous(
+              panGesture,
+              blockScrollUntilAtTheTop
+            )}>
```
Comment (j-piasecki) [inline examples/Example/src/new_api/bottom_sheet/index.tsx:129]: Here the `blockScrollUntilAtTheTop` gesture is attached to both gestures detectors. To just mark a relation without attaching it you can use `simultaneousWithExternalGesture`.

**Review by j-piasecki — COMMENTED** [review]
If I recall correctly this example requires both https://github.com/software-mansion/react-native-gesture-handler/pull/1611 and https://github.com/software-mansion/react-native-gesture-handler/pull/1609 for calculating gesture translations relative to the window and not the view. It should be merged when they are merged (or at least the part that changes the way translations are calculated).

**Review by j-piasecki — APPROVED** [review]

**Comment by victorzimnikov** [comment]
blockScrollUntilAtTheTop not working for android

**Pull request merged**